### DISCUSSION TITLE
rabbitmq   use haproxy vip

### DISCUSSION THREAD

**Discussion by emg-sys** [rabbitmq/rabbitmq-server#6405] [Other]
3 server rabbitmq queue cluster mirroring mode, openstack uses haproxy vip to access the rabbitmq queue, has this scenario been tested? Sometimes the production environment cannot be accessed through vip, what is the reason? There is no error in the openstack log, and no error in the rabbitmq queue.

**Comment by mkuratczyk** [comment]
I don't know about Openstack, but we have many users with HAProxy in front
of RabbitMQ clusters and no issues.

Best,

On Thu, Nov 10, 2022 at 11:20 AM emg-sys ***@***.***> wrote:

> 3 server rabbitmq queue cluster mirroring mode, openstack uses haproxy vip
> to access the rabbitmq queue, has this scenario been tested? Sometimes the
> production environment cannot be accessed through vip, what is the reason?
> There is no error in the openstack log, and no error in the rabbitmq queue.
>
> —
> Reply to this email directly, view it on GitHub
> <https://github.com/rabbitmq/rabbitmq-server/issues/6404>, or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ACI7PIQCXPFASIJK7ZEX7CDWHTEAVANCNFSM6AAAAAAR4MNOJQ>
> .
> You are receiving this because you are subscribed to this thread.Message
> ID: ***@***.***>
>


-- 
Michał
RabbitMQ team

**Comment by michaelklishin** [comment]
Consider using Discussions and not issues in repositories that have Discussions enabled.

Some side effects of proxies are [documented in the Heartbeats guide](https://www.rabbitmq.com/heartbeats.html#tcp-proxies). Others are: if there is only one proxy, it become a bandwidth bottleneck.

And, as you now see first hand, it can make [troubleshooting](https://www.rabbitmq.com/troubleshooting-networking.html) of the overall system harder. HAproxy has
logs and a UI with metrics. Start with inspecting those. RabbitMQ logs [all connection attempts that send at least one byte of data](https://rabbitmq.com/logging.html#connection-lifecycle-events).
